watchtower/wtclient/client.go

```diff
@@ -42,12 +42,22 @@ const (
 	DefaultForceQuitDelay = 10 * time.Second
 )
 
-// genActiveSessionFilter generates a filter that selects active sessions that
-// also match the desired channel type, either legacy or anchor.
-func genActiveSessionFilter(anchor bool) func(*ClientSession) bool {
-	return func(s *ClientSession) bool {
-		return s.Status == wtdb.CSessionActive &&
-			anchor == s.Policy.IsAnchorChannel()
+// genSessionFilter constructs a filter that can be used to select sessions only
+// if they match the policy of the client. If activeOnly is set, then only
+// active sessions will be returned.
+func (c *TowerClient) genSessionFilter(
+	activeOnly bool) wtdb.ClientSessionFilterFn {
+
+	return func(session *wtdb.ClientSession) bool {
+		if c.cfg.Policy.TxPolicy != session.Policy.TxPolicy {
```

Comment:
nit: discussed this on chat, so just taking note so we don't forget. The change in the filter is not super clear on first read, maybe consider separating to a new commit or add some more comment here.

Reply:
good catch - reverted it to just checking "isAnchor" and "active" for now. Can change the logic of the filter in a different PR 👍